apps/wallet-mobile/src/features/Receive/useCases/ListMultipleAddressesScreen.tsx

```diff
@@ -7,78 +7,81 @@ import {SafeAreaView} from 'react-native-safe-area-context'
 import {Button, Spacer, useModal} from '../../../components'
 import {useSelectedWallet} from '../../../SelectedWallet'
 import {BIP32_HD_GAP_LIMIT} from '../common/contants'
-import {useMultipleAddressesModal} from '../common/multipleAddressesModal'
 import {useReceive} from '../common/ReceiveProvider'
 import {ShowAddressLimitInfo} from '../common/ShowAddressLimitInfo/ShowAddressLimitInfo'
 import {SmallAddressCard} from '../common/SmallAddressCard/SmallAddressCard'
-import {useAddresses} from '../common/useAddresses'
+import {useAddressDerivationManager} from '../common/useAddressDerivationManager'
+import {useMultipleAddressesInfo} from '../common/useMultipleAddressesInfo'
 import {useNavigateTo} from '../common/useNavigateTo'
+import {useReceiveAddressesStatus} from '../common/useReceiveAddressesStatus'
 import {useStrings} from '../common/useStrings'
 import {QRs} from '../illustrations/QRs'
 
 type AddressInfo = {
-  isUsed: boolean
+  isUsed?: boolean
   address: string
 }
 
-export const MultipleAddressesScreen = () => {
+export const ListMultipleAddressesScreen = () => {
   const strings = useStrings()
   const {styles} = useStyles()
-  const addresses = useAddresses()
-  const {currentAddressChanged} = useReceive()
   const wallet = useSelectedWallet()
-
-  const {modalInfo} = useMultipleAddressesModal()
-  const mappedAddresses = mapAddresses(addresses)
   const navigate = useNavigateTo()
 
-  const {openModal} = useModal()
+  const {addressDerivation} = useAddressDerivationManager()
+  const addresses = useReceiveAddressesStatus(addressDerivation)
+  const {selectedAddressChanged} = useReceive()
+  React.useEffect(() => {
+    wallet.generateNewReceiveAddressIfNeeded()
+  }, [wallet])
 
+  const {openModal} = useModal()
+  const {isShowingMultipleAddressInfo} = useMultipleAddressesInfo()
   React.useEffect(() => {
-    modalInfo === false && openModal(strings.multiplePresentation, <Modal />, modalHeight)
-  }, [modalInfo, openModal, strings.multiplePresentation])
+    isShowingMultipleAddressInfo && openModal(strings.multiplePresentation, <Modal />, modalHeight)
+  }, [isShowingMultipleAddressInfo, openModal, strings.multiplePresentation])
 
-  const renderItem = React.useCallback(
+  const addressInfos = toAddressInfos(addresses)
+  const hasReachedGapLimit = addresses.unused.length >= BIP32_HD_GAP_LIMIT
+
+  const renderAddressInfo = React.useCallback(
     ({item}: {item: AddressInfo}) => (
       <SmallAddressCard
         address={item.address}
         isUsed={item.isUsed}
         onPress={() => {
-          currentAddressChanged(item?.address)
-          navigate.receiceDetails()
+          selectedAddressChanged(item.address)
+          navigate.receiveDetails()
         }}
-        // date={mocks.usedAddressDate}  // TODO don't have the date??
+        // date={}  // TODO define with project
```

Comment:
needs checking, we should drop it for the 1st iteration, it will require to scrap all txs. cc: @SorinC6